app/file_test.go

```diff
@@ -51,60 +46,57 @@ func TestDoUploadFile(t *testing.T) {
 	data := []byte("abcd")
 
 	info1, err := th.App.DoUploadFile(time.Date(2007, 2, 4, 1, 2, 3, 4, time.Local), teamId, channelId, userId, filename, data)
-	if err != nil {
-		t.Fatal(err)
-	} else {
+	require.Nil(t, err, "DoUploadFile should succeed with valid data")
+
+	if err == nil {
```

Comment:
As the `require.Nil` for the error is going to stop the test on the previous check if the error is not nil, I think this `if err == nil` conditional is unnecessary and can be safely removed

Reply:
@mgdelacroix Thanks. Your suggestion does not include the deferred function, just the conditional statement, right?

Reply:
@mgdelacroix

Reply:
Yes, only the conditional statements